iib/web/s3_utils.py

```diff
@@ -0,0 +1,33 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+import logging
+
+import boto3
+
+log = logging.getLogger(__name__)
+
+
+def get_file_from_s3_bucket(file_key, bucket_name):
+    """
+    Get artifact file from AWS S3 bucket.
+
+    :param str file_key: the key of the file to be fetched from the S3 bucket
+    :param str bucket_name: the name of the S3 bucket to fetch the file from
+    :return: the body of the S3 object or None
+    :rtype: botocore.response.StreamingBody
+    """
```

Comment:
Here `file_key` is the combination of `key_prefix` and `file_prefix` to fetch the object stored on  S3, right?  Something like `related_bundles/12345_related_bundles.json` for example.

I think it's a bit inconsistent to store the logs using `key_prefix and `file_prefix` then retrieving them by using a single parameter, can you please elaborate why are we doing this?

Finally, if we're using `conf['iib_aws_s3_bucket_name']` for the bucket, why do we need `bucket_name` as parameter?

Reply:
No specific reason, I can change it to accept two params. Good point!

Because we have `conf['iib_aws_s3_bucket_name']` in the worker config.

 And here, we have it in the Flask config. I can `import flask` and then use `flask.current_app.config['IIB_AWS_S3_BUCKET_NAME']` to fetch the bucket name but that would be unnecessary operation because we are already doing it in `get_request_logs` and `get_related_bundles` which are the functions calling get_file_from_s3_bucket. Hence, the bucket name as a parameter